### PR TITLE
ARROW-1438: [Python] Pull serialization context through PlasmaClient put and get

### PR DESCRIPTION
**Pull request by pcmoritz**

### CODE REVIEW

File: python/pyarrow/plasma.pyx [hunk 1]
```diff
@@ -370,7 +370,8 @@ cdef class PlasmaClient:
                                          object_buffers[i].metadata_size))
         return result
 
-    def put(self, object value, ObjectID object_id=None):
+    def put(self, object value, ObjectID object_id=None,
+            serialization_context=None):
```
Comment (xhochy) [inline python/pyarrow/plasma.pyx]: Could you the type to the new parameter?

**Review by xhochy — CHANGES_REQUESTED** [review]

File: python/pyarrow/plasma.pyx (hunk 1 above)
Comment (wesm) [reply python/pyarrow/plasma.pyx]: I believe it can be any Python object having the required methods

**Review by wesm — COMMENTED** [review]
Can you add a unit test?

File: python/pyarrow/plasma.pyx (hunk 1 above)
Comment (pcmoritz) [reply python/pyarrow/plasma.pyx]: Yes, this is true!

**Comment by pcmoritz** [comment]
Good catch, I'm adding the tests!

**Comment by pcmoritz** [comment]
+1 This should now be ready to merge

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**